src/lib/CLValue2/Abstract.ts

```diff
@@ -11,3 +15,19 @@ export abstract class CLValue {
 export abstract class ToBytes {
   toBytes: () => Uint8Array;
 }
+
+export abstract class FromBytes {
+  static fromBytes: (bytes: Uint8Array) => ResultAndRemainder<CLValue & ToBytes & FromBytes, CLErrorCodes>
```

Comment:
I don't want the concept of remainder to be our public API. From the point of the end users, they just want to check if all the bytes can be deserialized to given CLValue. I would rename this to `consume` and make module private (if possible) and `fromBytes` should return `Result<CLValue, CLErrorCodes>` if something happens or remainder is not empty.